### PR TITLE
Tesla Ball Fixes

### PR DESCRIPTION
**Pull request by TehFlaminTaco**
Tesla ball now varies in colour (Purples, Blues, And Whites) And changes
between them.
Tesla ball now actually stays in the centre, instead of drifting back to
x, y 0 0.
Tesla ball now has a ghost bias. Nothing overpowered, but if enough
players want the tesla ball to fuck something up, they can persuade it.

### CODE REVIEW

**Comment by ToGWtF** [comment]
I love this.

**Comment by DavidEdwards** [comment]
I am uncomfortable with ghosts fucking with the game world. I would run this by the council.

I like the other changes though.

**Comment by Cruix** [comment]
Make the ghost bias toggleable by admins. Even better, give it a weight toggleable by admins, so it can be ramped up for <s>illicit events</s> endround fun.

**Comment by TehFlaminTaco** [comment]
I'll make the Ghost Bias controllable then.

**Comment by ShadowDeath6** [comment]
holy shit, making the goofball controllable would be hilarious. :+1:

**Comment by TehFlaminTaco** [comment]
Hold on, Travis failed, Let me patch that shit.

**Comment by Cruix** [comment]
It's not clamp it's Clamp

**Comment by TehFlaminTaco** [comment]
Thanks bby. max(min(n, upper), lower) works too.

**Comment by DavidEdwards** [comment]
OK, this is great. Thank you. Lets give this a run on the server. I'm not thrilled to have it controllable by ghost army by default. However, 8 or more ghosts certainly makes it do-able.

File: code/modules/power/tesla/energy_ball.dm [hunk 1]
```diff
@@ -62,15 +68,23 @@ var/list/blacklisted_tesla_types = list(/obj/machinery/atmospherics,
...
 /obj/singularity/energy_ball/proc/move_the_basket_ball(var/move_amount)
-	for(var/i = 0, i < move_amount, i++)
-		var/move_dir = pick(alldirs)
-		var/turf/T = get_step(src,move_dir)
-		if(can_move(T))
-			loc = get_step(src,move_dir)
+	spawn(-1)
+		for(var/i = 0, i < move_amount, i++)
+			var/move_dir = pick(alldirs)
+			for (var/mob/dead/observer/ghost in range(21, get_turf(src)))
```
Comment (AsV9) [inline code/modules/power/tesla/energy_ball.dm:74]: Hahahaha

**Pull request merged**